.github/workflows/go.yml

```diff
@@ -15,6 +15,8 @@ jobs:
     strategy:
       matrix:
         go-version: ['1.22.x', '1.23.x']
+    env:
+      GO_NOT_LATEST: ${{ matrix.go-version != '1.23.x' }}
```

Comment:
The negative is a little confusing, but this is to avoid requiring special variables in local development -- the default in dev is to assume we're using the latest, which runs all tests.